python_modules/dagster/dagster/_core/definitions/time_window_partitions.py

```diff
@@ -154,14 +234,18 @@ def __new__(
                 # so we convert it to the provided timezone
                 start_dt = start_dt.in_tz(tz=timezone)
         else:
-            start_dt = pendulum.instance(datetime.strptime(start, fmt), tz=timezone)
+            start_dt = dst_safe_strptime(start, timezone, fmt)
 
         if not end:
             end_dt = None
         elif isinstance(end, datetime):
             end_dt = pendulum.instance(end, tz=timezone)
+            if end.tzinfo:
+                # Pendulum.instance does not override the timezone of the datetime object,
+                # so we convert it to the provided timezone
+                end_dt = end_dt.in_tz(tz=timezone)
```

Comment:
in_tz actually has DST transition issues too, but only in pendulum *2*, believe it or not. There's a to_timezone method that works around that: https://github.com/dagster-io/dagster/blob/master/python_modules/dagster/dagster/_seven/compat/pendulum.py#L57, we should probably use that for both the start_dt and the end_dt (and should probably have a lint rule to catch new callsites, looks like these two are the only ones currently)

Reply:
updated 👍